### PR TITLE
Fix the potential issues from a static field

### PR DESCRIPTION
**Pull request by LinuxSuRen**
A static final field should not be changed.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

**Comment by timja** [comment]
I don’t understand this? Could you explain more please

**Comment by LinuxSuRen** [comment]
The following `EMPTY` might not be empty. If someone calls the `put` method.
`public static final Mapping EMPTY = new Mapping();`

For example:
```
Mapping.EMPTY.put("fake", "fake");
```

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java [hunk 1]
```diff
@@ -74,4 +73,8 @@ public Mapping clone() {
         });
         return clone;
     }
+
+    public static Mapping getEmpty() {
```
Comment (timja) [inline plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: I would just call it empty, you aren’t getting a field on a specific instance

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java [hunk 2]
```diff
@@ -9,7 +9,6 @@
 
 public final class Mapping extends HashMap<String, CNode> implements CNode {
 
-    public static final Mapping EMPTY = new Mapping();
```
Comment (timja) [inline plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Breaking change?

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java (hunk 1 above)
Comment (LinuxSuRen) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Good idea.

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java (hunk 2 above)
Comment (LinuxSuRen) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: How about keep it. And mark as deprecated.

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java [hunk 3]
```diff
@@ -9,6 +9,8 @@
 
 public final class Mapping extends HashMap<String, CNode> implements CNode {
 
+    // Deprecated by PR #1240
+    public static final Mapping EMPTY = new Mapping();
```
Comment (timja) [inline plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Add the deprecated annotation

Comment (LinuxSuRen) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Done

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Please use proper deprecation annotation:
https://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html

Comment (timja) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: I assume you mean add the java doc deprecation to document what to use instead?

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]: Yes :)

File: plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java [hunk 4]
```diff
@@ -9,6 +9,8 @@
 
 public final class Mapping extends HashMap<String, CNode> implements CNode {
 
+    // Deprecated by PR #1240
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/model/Mapping.java]:
```suggestion
    /**
     * @deprecated replaced by {@link #empty()}
     */
    @Deprecated
```

**Comment by casz** [comment]
Nah, the annotation gotta stay too 😭

**Comment by LinuxSuRen** [comment]
> Nah, the annotation gotta stay too 😭
> https://stackoverflow.com/a/9031854

Got it. Thanks for the link.

File: plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
     private static final Logger logger = Logger.getLogger(HudsonPrivateSecurityRealmConfigurator.class.getName());
     private static final VersionNumber MIN_VERSION_FOR_HASHED_PASSWORDS = new VersionNumber("2.161");
     /// matches HudsonPrivateSecurityRealm.JBCRYPT_HEADER
-    private static final String HASHED_PASSWORD_PREFIX = "#jbcrypt:";
+    private static final String HASHED_PASSWORD_PREFIX = "#jbcrypt:"; // Noncompliant
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: not sure why this comment is needed?

Comment (LinuxSuRen) [reply plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: Sonarqube will detect it, and report it as a security issue due to keyword `PASSWORD`.

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: would be great to have sonarcloud setup 😉

Comment (LinuxSuRen) [reply plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: Agree. It'll be better if there're more automatic-check exists.

**Pull request closed without merging**